rbac/migration_tool/migrate.py

```diff
@@ -94,10 +97,52 @@ def migrate_role(
     return relationships, v2_role_bindings
 
 
-def migrate_users_for_groups(tenant: Tenant) -> list[common_pb2.Relationship]:
-    """Write users relationship to groups."""
+def migrate_system_role_assignments_for_groups(groups: list[Group]) -> list[common_pb2.Relationship]:
+    """Generate system role assignments for groups."""
+    with transaction.atomic():
+        dual_write_handler = None
+        relationships: list[common_pb2.Relationship] = []
+        for group in groups:
+            if dual_write_handler is None:
+                dual_write_handler = RelationApiDualWriteGroupHandler(group, None)
+            else:
+                dual_write_handler.group = group
+            system_roles = group.roles().filter(system=True)
+            if system_roles.exists() > 0:
+                dual_write_handler.generate_relations_to_add_roles(system_roles)
+                relationships.extend(dual_write_handler.group_relations_to_add)
+    return relationships
```

Comment:
yes I know that dual write handler is not designed for multiple groups and I wanted to avoid to do same query for default workspace for each group. Do you think it is fine with this fact ?

Reply:
I think this is fine, we can cache default workspace in follow up PR.